pkg/reconciler/revision/reconcile_resources.go

```diff
@@ -116,9 +116,13 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 						} else {
 							rev.Status.MarkContainerHealthyFalse(v1.ExitCodeReason(t.ExitCode), v1.RevisionContainerExitingMessage(t.Message))
 						}
-					} else if w := status.State.Waiting; w != nil && hasDeploymentTimedOut(deployment) {
-						logger.Infof("marking resources unavailable with: %s: %s", w.Reason, w.Message)
-						rev.Status.MarkResourcesAvailableFalse(w.Reason, w.Message)
+					} else if w := status.State.Waiting; w != nil {
+						if hasDeploymentTimedOut(deployment) {
+							logger.Infof("marking resources unavailable with: %s: %s", w.Reason, w.Message)
+							rev.Status.MarkResourcesAvailableFalse(w.Reason, w.Message)
+						} else {
+							rev.Status.PropagateDeploymentAvailabilityStatusIfFalse(&deployment.Status)
```

Comment:
The status message on minAvailable isn't very informative. In the reported issue I'm wondering if it's better to surface `ErrImagePull/ImagePullBackOff` from the Pod. 

I believe that's what's the original code intended - but `hasDeploymentTimedOut` is broken because it doesn't take into account a deployment not changing and scaling from zero. It seems like that's what we should be addressing here